upgrade.go

```diff
@@ -201,6 +201,34 @@ func (b *versionedKVBackend) Upgrade(ctx context.Context, s logical.Storage) err
 		return nil
 	}
 
+	prepareUpgradeInfoDoneFunc := func() ([]byte, error) {
+		upgradeInfo.Done = true
+		info, err := proto.Marshal(upgradeInfo)
+		if err != nil {
+			b.Logger().Error("encoding upgrade info resulted in an error", "error", err)
+			return nil, err
+		}
+		return info, nil
+	}
+
+	writeUpgradeInfoDoneFunc := func(info []byte) {
+	READONLY_LOOP:
+		for {
+			err = s.Put(ctx, &logical.StorageEntry{
+				Key:   path.Join(b.storagePrefix, "upgrading"),
+				Value: info,
+			})
+			switch {
+			case err == nil:
+				break READONLY_LOOP
```

Comment:
You could get rid of the label and just return here instead.

Reply:
Good point - updating accordingly